adagrams/game.py

```diff
@@ -1,11 +1,114 @@
+import random
+
 def draw_letters():
-    pass
+
+    letter_probability = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+    }
+    
+    # Create pool of letters with frequency as outlined in letter_probability dict
+    pool_of_letters = []
+    for letter in letter_probability:
+        pool_of_letters.extend([letter] * letter_probability[letter])
+
+    # Choose a hand of 10 letters
+    hand_of_letters = []
+
+    while len(hand_of_letters) < 10:
+        random_index = random.randint(0, len(pool_of_letters) - 1)
+        random_letter = pool_of_letters[random_index]
+
+    # Ensure no letter is overrepresented compared to letter_probability dict
+        if hand_of_letters.count(random_letter) < letter_probability[random_letter]:
+            hand_of_letters.append(pool_of_letters[random_index])
+
+    return hand_of_letters
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    word = word.upper()
+
+    # Ensure that letters in word are available in correct quantities in hand
+    for letter in word:
+        if letter not in letter_bank or (word.count(letter) > letter_bank.count(letter)):
```

Comment:
This is another scenario in which mutation is avoided at the expense of a complex condition. I think it's simpler to create a shallow copy of `letter_bank`, then check whether the letter is in the word, removing it from the letter bank copy if so (if not, we've encountered a letter not in the letter bank, so we can return false).
```python
    letter_bank_copy = letter_bank[:]
    for letter in word:
        if letter in letter_bank_copy:
            letter_bank_copy.remove(letter)
        else:
            return False

    return True
```